core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -51,6 +58,7 @@ class ControllerContext(val zkUtils: ZkUtils) {
   var partitionLeadershipInfo: mutable.Map[TopicAndPartition, LeaderIsrAndControllerEpoch] = mutable.Map.empty
   val partitionsBeingReassigned: mutable.Map[TopicAndPartition, ReassignedPartitionsContext] = new mutable.HashMap
   val partitionsUndergoingPreferredReplicaElection: mutable.Set[TopicAndPartition] = new mutable.HashSet
+  val replicasOnOfflineDisks: mutable.Map[Int, Set[TopicPartition]] = mutable.HashMap.empty
```

Comment:
I was thinking that we may want to use the Java version for new code so that we can gradually migrate to the Java version. I have change the code to use TopicAndPartition.